Shared/Notifications/NotificationsManager.swift

```diff
@@ -0,0 +1,209 @@
+                            if itemServingsByMealAndDate[menuItemName]![day] == nil {
+                                itemServingsByMealAndDate[menuItemName]![day] = [(String, Date, CampusEatery.EventName)]()
+                            }
+                            itemServingsByMealAndDate[menuItemName]![day]?.append((eatery: eatery.displayName, eventStart: event.start, eventName: eventName))
+                        }
+                    }
+                }
+            }
+                  
+            // Iterate through all servings of all favorite menu items on all days
+            for (menuItemName, daysAndServings) in itemServingsByMealAndDate {
+                // Iterate through all servings of menuItemName on each day it is served
+                for (_, itemServings) in daysAndServings {
+                    // Groups menuItemName servings by which event name they're contained in
+                    var servingsByEventName = [CampusEatery.EventName : [(eatery: String, eventStart: Date)]]()
+                    for serving in itemServings {
+                        if servingsByEventName[serving.eventName] == nil {
+                            servingsByEventName[serving.eventName] = [(String, Date)]()
+                        }
+                        servingsByEventName[serving.eventName]!.append((serving.eatery, serving.eventStart))
+                    }
+                    // Iterate through menuItemName servings by which event name they're contained in, on a specific day
+                    for (eventName, servings) in servingsByEventName {
+                        // Sort servings on a specific event day and date and then store the eateries at which such servings take place, in increasing order of event start
+                        var sortedServings = servings
+                        sortedServings.sort { $0.eventStart < $1.eventStart }
+                        let eateryNames = sortedServings.map { $0.eatery }
+                              
+                        // Set the notification date to be one hour before the event start of the first serving at  any eatery on day and eventName
+                        let numSecondsInHour: TimeInterval = 60 * 60
+                        let notificationDate = sortedServings[0].eventStart.addingTimeInterval(-numSecondsInHour)
+                        
+                        var requiresNotifScheduling = true
+                        // Stores scheduled notification identifiers that are to be unscheduled
+                        var disposableNotifIdentifiers = [String]()
+                        // Iterate through current scheduled notifications, removing any scheduled notifications with outdated information, keeping notifications that match the most recent menu information, and adding notifications if an oudated notification existed or no notification existed
+                        if let scheduledMenuItemNotifs = self.scheduledNotifications[menuItemName] {
```

Comment:
Style: Use `guard` statements here to reduce the levels of nesting

Reply:
The guard logic doesn't seem to make sense here since these are all things to be done given a condition, but things after this code don't depend on that condition, and there is nothing happening explicitly when the condition is false.